sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py

```diff
@@ -4039,3 +4177,62 @@ def from_dict(cls, data):
             innererror=DocumentAnalysisInnerError.from_dict(data.get("innererror"))  # type: ignore
             if data.get("innererror") else None
         )
+
+def _get_children(source_element, search_elements, allowed_elements, cross_page=False):
+    # Check if the element types that are passed in are allowed,
+    # fail fast if an incorrect element is requested.
+    for elem in search_elements:
+        if elem not in allowed_elements:
+            raise ValueError("Received an unsupported value in element_types")
+    result = []
+
+    # search for elements across pages if cross_page is set to True
+    if cross_page:
+        for elem in search_elements:
+            for region in source_element.bounding_regions:
+                for element in _get_element_list(_get_page(source_element._parent.pages, region.page_number), elem):
+                    if _in_span(element, source_element.spans):
+                        result.append(element)
```

Comment:
Idk where the other comment went so I'll say it here - We set _parent to the generated AnalyzeResult/etc model in the _from_generated methods. In _get_children we append the elements to the result list from the _parent so the lines and words returned are actually those of the generated type. We should do a quick transform to the exposed models before we return to the user. If not, any generated DocumentLines returned from get_lines() won't have a get_words() on it.

Reply:
Ah I see what you're saying, I hadn't understood your first comment about this. Yes you're right, we should transform before returning. Will do.